services/skus/service.go

```diff
@@ -2488,3 +2497,84 @@ func shouldUpdateOrderStripeSubID(ord *model.Order, subID string) bool {
 
 	return false
 }
+
+func shouldTransformStripeOrder(ord *model.Order) bool {
+	if ord.IsIOS() {
+		return false
+	}
+
+	if ord.IsAndroid() {
+		return false
+	}
+
+	return !ord.IsPaid() && ord.IsStripePayable()
+}
+
+func chooseStripeSessID(ord *model.Order, canBeNewSessID string) (string, bool) {
+	if canBeNewSessID != "" {
+		return canBeNewSessID, true
+	}
+
+	return ord.StripeSessID()
+}
+
+type createStripeSessionRequest struct {
+	orderID    string
+	email      string
+	successURL string
+	cancelURL  string
+	trialDays  int64
+	items      []*stripe.CheckoutSessionLineItemParams
+}
+
+func createStripeSession(ctx context.Context, cl stripeClient, req createStripeSessionRequest) (string, error) {
+	params := &stripe.CheckoutSessionParams{
+		PaymentMethodTypes: []*string{ptrTo("card")},
+		Mode:               ptrTo(string(stripe.CheckoutSessionModeSubscription)),
+		SuccessURL:         &req.successURL,
+		CancelURL:          &req.cancelURL,
+		ClientReferenceID:  &req.orderID,
+		SubscriptionData:   &stripe.CheckoutSessionSubscriptionDataParams{},
+		LineItems:          req.items,
+	}
+
+	if custID, ok := cl.FindCustomer(ctx, req.email); ok {
+		params.Customer = &custID.ID
+	} else {
+		if req.email != "" {
+			params.CustomerEmail = &req.email
+		}
+	}
+
+	if req.trialDays > 0 {
+		params.SubscriptionData.TrialPeriodDays = &req.trialDays
+	}
+
+	params.SubscriptionData.AddMetadata("orderID", req.orderID)
+	params.AddExtra("allow_promotion_codes", "true")
+
+	sess, err := cl.CreateSession(ctx, params)
+	if err != nil {
+		return "", err
+	}
+
+	return sess.ID, nil
+}
+
+func buildStripeLineItems(items []model.OrderItem) []*stripe.CheckoutSessionLineItemParams {
+	var result []*stripe.CheckoutSessionLineItemParams
+
+	for i := range items {
+		priceID, ok := items[i].StripeItemID()
+		if !ok {
+			continue
```

Comment:
question: is there a reason we continue processing when there is no `priceID`, would this not be an error?

Reply:
At this point it's not an issue because there always is one item, and other validation steps check that there is the metadata. Will reconsider it later when working on bundles.

Reply:
Resolving for now.